packages/react-dom/src/__tests__/ReactEventIndependence-test.js

```diff
@@ -10,74 +10,56 @@
 'use strict';
 
 let React;
-let ReactDOM;
+let ReactDOMClient;
+let act;
 
 describe('ReactEventIndependence', () => {
   beforeEach(() => {
     jest.resetModules();
 
     React = require('react');
-    ReactDOM = require('react-dom');
+    ReactDOMClient = require('react-dom/client');
+    act = require('internal-test-utils').act;
   });
 
-  it('does not crash with other react inside', () => {
+  it('does not crash with other react inside', async () => {
     let clicks = 0;
     const container = document.createElement('div');
     document.body.appendChild(container);
+    const root = ReactDOMClient.createRoot(container);
     try {
-      const div = ReactDOM.render(
-        <div
-          onClick={() => clicks++}
-          dangerouslySetInnerHTML={{
-            __html: '<button data-reactid=".z">click me</div>',
-          }}
-        />,
-        container,
-      );
+      await act(() => {
+        root.render(
+          <div
+            onClick={() => clicks++}
+            dangerouslySetInnerHTML={{
+              __html: '<button data-reactid=".z">click me</div>',
+            }}
+          />,
+        );
+      });
 
-      div.firstChild.click();
+      container.firstElementChild.click();
       expect(clicks).toBe(1);
     } finally {
       document.body.removeChild(container);
     }
   });
 
-  it('does not crash with other react outside', () => {
+  it('does not crash with other react outside', async () => {
     let clicks = 0;
     const outer = document.createElement('div');
     document.body.appendChild(outer);
+    const root = ReactDOMClient.createRoot(outer);
     try {
       outer.setAttribute('data-reactid', '.z');
-      const inner = ReactDOM.render(
-        <button onClick={() => clicks++}>click me</button>,
-        outer,
-      );
-      inner.click();
+      await act(() => {
+        root.render(<button onClick={() => clicks++}>click me</button>);
+      });
+      outer.firstElementChild.click();
       expect(clicks).toBe(1);
     } finally {
       document.body.removeChild(outer);
     }
   });
-
-  it('does not when event fired on unmounted tree', () => {
-    let clicks = 0;
-    const container = document.createElement('div');
-    document.body.appendChild(container);
-    try {
-      const button = ReactDOM.render(
-        <button onClick={() => clicks++}>click me</button>,
-        container,
-      );
-
-      // Now we unmount the component, as if caused by a non-React event handler
-      // for the same click we're about to simulate, like closing a layer:
-      ReactDOM.unmountComponentAtNode(container);
```

Comment:
Since `unmountComponentAtNode` is deprecated and `root.unmount()` completely removes the button element, I'm not sure there's anything else to test here so I've removed the test case. Any reason to continue testing events on unmounted trees like this?

Reply:
@jackpope `root.unmount()` doesn't remove the container, which is where the event listeners are attached so the test is needed to confirm the event handlers on the container are not fired.

Reply:
Trying node#remove because I don't think we have any non-deprecated APIs that can remove the container?

If I remove the container from the DOM, the click handler still fires.
```javascript
container.remove();
button.click();
expect(clicks).toBe(0); // assertion fails
```
If I remove the button from the DOM, the event handler does not fire

```javascript
button.remove();
button.click();
expect(clicks).toBe(0);
```

Is this expected?

Reply:
Sorry, I'm confused, why not use `root.unmount()`? That behaves the same as `unmountComponentAtNode` which also removes the button element?

Reply:
It does seem silly though to confirm that when you remove an element from the DOM, clicking doesn't bubble up to the element it was attached to. The test is really old, @sophiebits do you know what this is testing for and if it's still relevant?

Reply:
Maybe it pre-dates event delegation, which is funny because it will be relevant again with the event system changes we have planned lol

Reply:
Oh, maybe this is from when we still delegated events to the document, since detached elements still fire events on the document? Which is interesting because that would make it not relevant post-17, but will be relevant again after the event changes we want to make?

Reply:
You can read the original context here: https://github.com/facebook/react/issues/4865 https://github.com/facebook/react/issues/3790

Reply:
@rickhanlonii `root.unmount()` nullifies any reference I've tried to store for the button element. So I haven't been able to click on anything. Unlike `unmountComponentAtNode` which clears the container in the same way but preserves the button element reference.

Reply:
@jackpope this works:

```js
it('does not when event fired on unmounted tree', async () => {
    let clicks = 0;
    const container = document.createElement('div');
    document.body.appendChild(container);
    try {
      const root = ReactDOMClient.createRoot(container);

      await act(async () => {
        root.render(<button onClick={() => clicks++}>click me</button>);
      });

      const button = container.firstChild;

      // Now we unmount the component, as if caused by a non-React event handler
      // for the same click we're about to simulate, like closing a layer:
      root.unmount();
      button.click();

      // Since the tree is unmounted, we don't dispatch the click event.
      expect(clicks).toBe(0);
    } finally {
      document.body.removeChild(container);
    }
  });
  ```

Reply:
Ah so I was overcomplicating it. Thanks!